lib/core/util.js

```diff
@@ -170,6 +170,37 @@ function isBuffer (buffer) {
   return buffer instanceof Uint8Array || Buffer.isBuffer(buffer)
 }
 
+function mismatchContentLength (bodySize, contentLength) {
+  return bodySize > 0 && contentLength > 0 &&
+    bodySize !== contentLength
+}
+
+function extractMeaningfulHeaders (rawHeaders) {
+  const headers = {
+    keepAlive: null,
+    trailers: null,
+    contentLength: 0
+  }
```

Comment:
I'm a little worried this extra object has perf implications? Can't we just send in the parser as an arg and set them direcrtly?

Reply:
I don't see much difference between instancing 3 vars instead of one object, btw the destructuration of the result is not needed, so I removed

Reply:
I'm ok to move back without this function

Reply:
It would be better if we could avoid it.

Reply:
removed